test/Core.Test/Services/OrganizationServiceTests.cs

```diff
@@ -138,5 +146,216 @@ public async Task OrgImportCreateNewUsersAndMarryExistingUser()
             await orgUserRepo.Received(1).UpsertAsync(Arg.Any<OrganizationUser>());
             await orgUserRepo.Received(2).CreateAsync(Arg.Any<OrganizationUser>());
         }
+
+        [Theory, CustomAutoDataAttribute(typeof(SutProviderCustomization))]
+        public async Task UpgradePlan_OrganizationIsNull_Throws(Guid organizationId, OrganizationUpgrade upgrade,
+                SutProvider<OrganizationService> sutProvider)
+        {
+            sutProvider.GetDependency<IOrganizationRepository>().GetByIdAsync(organizationId).Returns(Task.FromResult<Organization>(null));
+            var exception = await Assert.ThrowsAsync<NotFoundException>(
+                () => sutProvider.Sut.UpgradePlanAsync(organizationId, upgrade));
+        }
+
+        [Theory, CustomAutoDataAttribute(typeof(SutProviderCustomization))]
+        public async Task UpgradePlan_GatewayCustomIdIsNull_Throws(Organization organization, OrganizationUpgrade upgrade,
+                SutProvider<OrganizationService> sutProvider)
+        {
+            organization.GatewayCustomerId = string.Empty;
```

Comment:
What if a GatewayCustomerId exists but is not valid for some reason (other than null or empty?)

Reply:
There is no other validation checked on that field. An error throws if it doesn't exist, but as far as I can tell any other string is fair game here.

Reply:
Is that the desired behavior? I don't know the code so maybe this isn't the part that should deal with it, but it seems like upgrade should be unsuccessful in some way if you can't actually charge for it